api/src/main/java/org/jboss/forge/roaster/model/util/Types.java

```diff
@@ -298,7 +298,12 @@ public static boolean isGeneric(String type)
       StringTokenizer tok = new StringTokenizer(typeArgs, ", ");
       while (tok.hasMoreTokens())
       {
-         if (!validateNameWithGenerics(tok.nextToken()))
+         String typeArg = tok.nextToken();
+         while (incompleteGenerics(typeArg) && tok.hasMoreElements()) {
+            typeArg += tok.nextToken();
```

Comment:
If my new method accepts StringBuilder, it would have to call `toString()` in order to check it. This would happen in each iteration, so there is no performance improvement.